background/services/internal-ethereum-provider/index.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  ChainService,
+  ServiceCreatorFunction,
+  ServiceLifecycleEvents,
+} from ".."
+import BaseService from "../base"
+
+type Events = ServiceLifecycleEvents & {
+  // connect
+  // disconnet
+  // account change
+  // networkchange
+}
+
+export default class InternalEthereumProviderService extends BaseService<Events> {
+  static create: ServiceCreatorFunction<
+    ServiceLifecycleEvents,
```

Comment:
```suggestion
    Events,
```

Just to make sure we don't lose the provider-specific events in the types and get confused later :)

Reply:
Thanks 😌